### PR TITLE
[cmd/githubgen] add distribution reports

### LINKED ISSUE #28628: Distro should include owners
### Component(s)

cmd/mdatagen

### Describe the issue you're reporting

There's a list of distros available in mdatagen:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5e285f0ebdb35e103b292a232fa65910f4c9c879/cmd/mdatagen/statusdata.go#L14-L24

However, this list doesn't include any owners. This makes it difficult to review PRs that include adding a component to any particular distro. I propose that owners should be listed, much like code owners for each of the listed distro.

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Adds a set of distribution reports that can be used to notify distribution maintainers of any changes to distributions.

**Link to tracking Issue:**
Fixes #28628

### CODE REVIEW

File: reports/distributions/liatrio.yaml
Comment (mx-psi) [inline reports/distributions/liatrio.yaml:6]: It looks odd to me that a distribution has a single component

File: exporter/azuremonitorexporter/trace_to_envelope.go
Comment (mx-psi) [inline exporter/azuremonitorexporter/trace_to_envelope.go:303]: This change and the other one in azuremonitorexporter are not related to the PR AFAICT, could we remove them?

File: cmd/githubgen/main.go
Comment (mx-psi) [inline cmd/githubgen/main.go:41]: Comment above needs an update about what other files we update with this

File: cmd/githubgen/allowlist.txt
Comment (mx-psi) [inline cmd/githubgen/allowlist.txt:24]: This does not look related to the PR changes, can we move it to a different PR if so? (Also @braydonk happy to sponsor your OTel membership :))

File: .github/workflows/scripts/check-codeowners.sh
Comment (mx-psi) [inline .github/workflows/scripts/check-codeowners.sh:66]: Why do we need this change? Also sounds like a candidate for a separate PR :smile:

File: .github/CODEOWNERS
Comment (mx-psi) [inline .github/CODEOWNERS:24]: How are the spacing changes related to adding the distributions report? Are they necessary?

File: .github/workflows/scripts/check-codeowners.sh
Comment (atoulme) [reply .github/workflows/scripts/check-codeowners.sh:66]: it's because the codeowners file validation fails if the file path offered is not a folder right now. Yes, I'll get an other PR for it, good point.

File: .github/CODEOWNERS
Comment (atoulme) [reply .github/CODEOWNERS:24]: no, they're not. Unfortunately the file was corrupted.

Comment (atoulme) [reply .github/CODEOWNERS:24]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29727 is open to help fix some of this

File: reports/distributions/liatrio.yaml
Comment (atoulme) [reply reports/distributions/liatrio.yaml:6]: heh well it might be so much value that's all they need.

File: exporter/azuremonitorexporter/trace_to_envelope.go
Comment (atoulme) [reply exporter/azuremonitorexporter/trace_to_envelope.go:303]: oh intellij was too eager in renaming, sorry, I'll revert.

File: cmd/githubgen/allowlist.txt
Comment (atoulme) [reply cmd/githubgen/allowlist.txt:24]: yes, it's now in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29727

File: .github/workflows/scripts/check-codeowners.sh
Comment (atoulme) [reply .github/workflows/scripts/check-codeowners.sh:66]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29737

File: cmd/githubgen/allowlist.txt
Comment (atoulme) [reply cmd/githubgen/allowlist.txt:24]: thanks! Resolved

File: .github/CODEOWNERS
Comment (atoulme) [reply .github/CODEOWNERS]: should be good there now, thanks!

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go:42]: thanks, added

**Comment by mx-psi** [comment]
Thanks for splitting out the PRs, should be good now if you rebase :)

File: cmd/githubgen/allowlist.txt
Comment (braydonk) [reply cmd/githubgen/allowlist.txt:24]: @mx-psi I will message you separately on Slack!

File: reports/distributions/liatrio.yaml
Comment (mx-psi) [reply reports/distributions/liatrio.yaml:6]: @adrielp Does liatrio's distro only have one component?

Comment (adrielp) [reply reports/distributions/liatrio.yaml:6]: Nope @mx-psi, you're spot on. I need to come back and updated this. This is what we include [taken from this manifest.yaml](https://github.com/liatrio/liatrio-otel-collector/blob/main/config/manifest.yaml) (i've updated the distro faster than I could open pr's for sure) 

Extensions:
- zpages
- ballast
- basicauth
- healthcheck
- pprof
- bearertokenauth
- sigv4auth

exporters:
- logging
- otlphttp
- awsxray
- awscloudwatchlogs
- influxdb
- prometheus
- zipkiin
- prometheusremotewrite

processors:
- batchprocessor
- memorylimiter
- resourcedetection
- resource
- k8sattributes
- transform
- filter

receivers:
- otlp
- gitprovider
- prometheus
- kubeletestats
- k8scluster
- jaeger
- zipkin

connectors:
- spanmetrics

Do you want me to make a suggestion to this file to add them?

Comment (atoulme) [reply reports/distributions/liatrio.yaml:6]: you need to add this to the metadata.yaml of every component you include, and then this tool will pick up the changes and build the report.
This is not blocking this PR.

**Review by mx-psi — APPROVED** [review]

File: reports/distributions/liatrio.yaml
Comment (adrielp) [reply reports/distributions/liatrio.yaml:6]: @atoulme  - want me to go ahead and do that now through a separate PR such that it can be picked up through this one?

Comment (atoulme) [reply reports/distributions/liatrio.yaml:6]: Please open a PR, and we'll see which one lands first.

**Review by jsirianni — COMMENTED** [review]
This is great. Left a couple comments related to the observIQ repository link.

File: distributions.yaml
Comment (atoulme) [reply distributions.yaml:14]: go ahead and open a PR to fix, and feel free to add yourself as maintainer so you can keep on top of this type of changes moving forward, thanks!

**Pull request merged**
